runtime/runtime.go

```diff
@@ -3163,11 +3153,6 @@ func NewPublicKeyValue(
 			getLocationRange func() interpreter.LocationRange,
 			publicKeyValue *interpreter.CompositeValue,
 		) interpreter.BoolValue {
-			// If the public key is already validated, avoid re-validating, and return the cached result.
-			if publicKey.Validated {
-				return interpreter.BoolValue(publicKey.IsValid)
-			}
-
```

Comment:
Is there a good way to avoid needing to re-validate now that there's no `isValid` field to act as a cache? I don't know the codebase well enough.

It might not be enough of a performance cost to worry about though.

Reply:
I think this was the main reason for having the `Validated` field of the internal public keys (Go type `PublicKey`, which is passed between Cadence and FVM). I'm not sure if we can assume that the keys from FVM are always valid. Maybe the PR should not get rid of the `Validated` field. 

Thoughts @tarakby ?

Reply:
parallel reviews! 😅

Reply:
😆

Reply:
If I understand correctly, this function is handling keys coming from FVM? 
If yes, keys coming from the FVM are always valid, there is no need to re-validate them. 

The `validated` field existed before to avoid re-validating the key when the `isValid` field is read on Cadence. That way validation is run only once per Cadence key and the result is cached in `isValid`.